src/util/mod.rs

```diff
@@ -110,3 +113,19 @@ impl From<network::Error> for Error {
         Error::Network(e)
     }
 }
+
+// core2 doesn't have read_to_end
+pub(crate) fn read_to_end<D: io::Read>(mut d: D) -> Result<Vec<u8>, io::Error> {
+    let mut result = vec![];
+    let mut buf = [0u8; 64];
+    loop {
+        match d.read(&mut buf) {
+            Ok(0) => break,
+            Ok(n) => result.extend_from_slice(&buf[0..n]),
+            #[cfg(feature = "std")]
+            Err(ref e) if e.kind() == ::std::io::ErrorKind::Interrupted => {},
```

Comment:
Why is this feature gated? [core2 seems to have a similar error kind](https://docs.rs/core2/0.3.1/core2/io/enum.ErrorKind.html#variant.Interrupted)

Reply:
Oh, right, I didn't consider that no_std could potentially handle interrupts.  Fixed.